### PR TITLE
test(delayedack): Add simulation tests for rollapp finalization

### LINKED ISSUE #995: Simulation tests for EndBlocker per-rollapp failure handling
In this issue https://github.com/dymensionxyz/dymension/issues/670 we have tackled the failure handling of rollapp packet finalization so that it is done per-rollapp.
Apart from the existing unit tests, the behavior needs to be tested across multiple blocks.

### PR DESCRIPTION
**Pull request by zale144**
## Description

----

Closes #995 

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----;

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

**Review by omritoptix — APPROVED** [review]

**Comment by mtsitrin** [comment]
tests fails

File: x/rollapp/keeper/block_height_to_finalization_queue_test.go [hunk 1]
```diff
@@ -81,137 +83,196 @@ func TestBlockHeightToFinalizationQueueGetAll(t *testing.T) {
...
 }
 
-// TODO: Test FinalizeQueue function with failed states
 func (suite *RollappTestSuite) TestFinalizeRollapps() {
-	type stateUpdate struct {
-		blockHeight int64
-		startHeight uint64
-		numBlocks   uint64
+	suite.SetupTest()
+
```
Comment (danwt) [inline x/rollapp/keeper/block_height_to_finalization_queue_test.go:88]: In testify I dont think you have to call this expliclity

**Review by danwt — CHANGES_REQUESTED** [review]
a test and golangcilint are failing

File: x/rollapp/keeper/block_height_to_finalization_queue_test.go (hunk 1 above)
Comment (zale144) [reply x/rollapp/keeper/block_height_to_finalization_queue_test.go:88]: you do

**Review by omritoptix — APPROVED** [review]

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**